runtime/oti/VMHelpers.hpp

```diff
@@ -2277,6 +2277,31 @@ class VM_VMHelpers
 		targetThread->privateFlags2 |= J9_PRIVATE_FLAGS2_REENTER_INTERPRETER;
 		indicateAsyncMessagePending(targetThread);
 	}
+
+
+	static U_32
+	setThreadState(J9VMThread *currentThread, U_32 state)
+	{
+		U_32 oldState = 0;
+#if JAVA_SPEC_VERSION >= 19
+		j9object_t receiverObject = currentThread->carrierThreadObject;
+		if (NULL != receiverObject) {
+			/* Platform threads must have a non-null FieldHolder object. */
+			j9object_t threadHolder = J9VMJAVALANGTHREAD_HOLDER(currentThread, receiverObject);
+			if (NULL != threadHolder) {
+				oldState = (U_32)J9VMJAVALANGTHREADFIELDHOLDER_THREADSTATUS(currentThread, threadHolder);
```

Comment:
Please remove the redundant casts here and on line 2299; notice `J9OBJECT_U32_LOAD` in
```c
#define J9VMJAVALANGTHREADFIELDHOLDER_THREADSTATUS(vmThread, object) ((void)0, \
        J9OBJECT_U32_LOAD(vmThread, object, J9VMJAVALANGTHREADFIELDHOLDER_THREADSTATUS_OFFSET(vmThread)))
```